### PR TITLE
[Patient Intake Example] Preferred Pharmacy

### LINKED ISSUE #5200: [Patient Intake Sample] Add preferred pharmacy
This is an FAQ and should be added here: https://github.com/medplum/medplum/tree/main/examples/medplum-patient-intake-demo

Related docs: https://www.medplum.com/docs/medications/representing-prescriptions-and-medication-orders#representing-dispense-instructions

> There is currently no community consensus on how to store a patient's preferred pharmacy in FHIR, but Mepdlum's guidance is to add this information to a [CareTeam](https://www.medplum.com/docs/api/fhir/resources/careteam) resource associated with the [Patient](https://www.medplum.com/docs/api/fhir/resources/patient).

### PR DESCRIPTION
**Pull request by pamella**
## Description

- Resolves #5200 
- Docs https://www.medplum.com/docs/medications/representing-prescriptions-and-medication-orders#representing-dispense-instructions
  > There is currently no community consensus on how to store a patient's preferred pharmacy in FHIR, but Mepdlum's guidance is to add this information to a [CareTeam](https://www.medplum.com/docs/api/fhir/resources/careteam) resource associated with the [Patient](https://www.medplum.com/docs/api/fhir/resources/patient).

## Steps to test

- [ ] Run `npm run build:bots`
- [ ] Upload data
  image
- [ ] Report preferred pharmacy when creating a new patient
- [ ] Check if the CareTeam resource have been created

[Screencast from 12-09-2024 12:39:01.webm](https://github.com/user-attachments/assets/727f8c96-3325-4e5f-a144-7eb53e0e7b50)

[Screencast from 12-09-2024 12:39:50.webm](https://github.com/user-attachments/assets/0c592249-26f1-4fac-99c4-9824059fc838)

### CODE REVIEW

File: examples/medplum-patient-intake-demo/data/example/example-organization-data.json [hunk 1]
```diff
@@ -59,6 +59,64 @@
...
+          {
+            "coding": [
+              {
+                "system": "http://terminology.hl7.org/CodeSystem/organization-type",
+                "code": "prov",
+                "display": "Healthcare Provider"
+              }
+            ]
+          }
+        ]
```
Comment (pamella) [inline examples/medplum-patient-intake-demo/data/example/example-organization-data.json:89]: `medplum-photon-integration` is using this same type.

File: examples/medplum-patient-intake-demo/src/bots/core/intake-utils.ts [hunk 2]
```diff
@@ -504,6 +505,42 @@ export async function addImmunization(
...
+      participant: [
+        {
+          member: pharmacy,
+          role: [{ coding: [{ system: SNOMED, code: '76166008', display: 'Practical aid (pharmacy) (occupation)' }] }],
+        },
+        {
+          member: createReference(patient),
+          role: [{ coding: [{ system: SNOMED, code: '116154003', display: 'Patient (person)' }] }],
+        },
+      ],
```
Comment (pamella) [inline examples/medplum-patient-intake-demo/src/bots/core/intake-utils.ts:535]: Based on https://build.fhir.org/ig/HL7/US-Core/StructureDefinition-us-core-careteam.html

image

File: examples/medplum-patient-intake-demo/src/bots/core/intake-utils.ts [hunk 3]
```diff
@@ -504,6 +505,42 @@ export async function addImmunization(
...
+      status: 'proposed',
+      name: 'Patient Preferred Pharmacy',
+      subject: createReference(patient),
+      participant: [
+        {
+          member: pharmacy,
+          role: [{ coding: [{ system: SNOMED, code: '76166008', display: 'Practical aid (pharmacy) (occupation)' }] }],
+        },
+        {
+          member: createReference(patient),
```
Comment (rahul1) [inline examples/medplum-patient-intake-demo/src/bots/core/intake-utils.ts:532]: Remove the reference to the paitent as a "participant". Typically, the participants are just the clinicians involved

**Review by rahul1 — APPROVED** [review]

File: examples/medplum-patient-intake-demo/src/bots/core/intake-utils.ts (hunk 3 above)
Comment (pamella) [reply examples/medplum-patient-intake-demo/src/bots/core/intake-utils.ts:532]: Oh, got it. Will do! Thanks

**Pull request merged**